src/fields.js

```diff
@@ -2544,88 +3347,296 @@ x3dom.fields.Quaternion.rotateFromTo = function ( fromVec, toVec )
+ *     </tr>
+ *   </thead>
+ *   <tbody>
+ *     <tr>
+ *       <td>color name</td>
+ *       <td>
+ *         One of the standard web color names as accepted by CSS as
+ *         a string value; for instance <code>"red"</code>
+ *       </td>
+ *     </tr>
+ *     <tr>
+ *       <td>
+ *         <code>rgb(<i>red</i>, <i>green</i>, <i>blue</i>)</code>
+ *       </td>
+ *       <td>
+ *         An invocation of the CSS-conformant function
+ *         <code>rgb()</code> with the three parameters red, green, and
+ *         blue, each of them being an integer in the range [0.0, 1.0].
+ *       </td>
+ *     </tr>
+ *     <tr>
+ *       <td>
+ *         <code>rgba(<i>red</i>, <i>green</i>, <i>blue</i>, <i>alpha</i>)</code>
+ *       </td>
+ *       <td>
+ *         An invocation of the CSS-conformant function
+ *         <code>rgba()</code> with the four parameters red, green,
+ *         blue, and alpha, the first three of which expected as an
+ *         integer in the range [0, 255], the fourth being a
+ *         floating-point number in [0.0, 1.0].
+ *       </td>
+ *     </tr>
+ *     <tr>
+ *       <td>hexadecimal integer value</td>
+ *       <td>
+ *         The representation of the color as a sharp sign
+ *         (<code>#</code) prefixed sequence of three to eight
+ *         hexadecimal digits, the four variants of which are these:
+ *         <ul>
+ *           <li>
+ *             <code>#RRGGBBAA</code> --- red, green, blue, and alpha;
+ *             each in a resolution of the integer range [0, 255]
+ *           </li>
+ *           <li>
+ *             <code>#RRGGBB</code> --- red, green, and blue; each in a
+ *             resolution of the integer range [0, 255]
+ *           </li>
+ *           <li>
+ *             <code>#RGBA</code> --- red, green, blue, and alpha;
+ *             each in a resolution of the integer range [0, 15]
+ *           </li>
+ *           <li>
+ *             <code>#RGB</code> --- red, green, and blue; each in a
+ *             resolution of the integer range [0, 15].
+ *           </li>
+ *         </ul>
+ *       </td>
+ *     </tr>
+ *   </tbody>
+ * </table>
+ *
+ * The thus generated object exhibits the following structure:
+ *   {
+ *     r : <red>,
+ *     g : <green>,
+ *     b : <blue>,
+ *     a : <alpha>
+ *   }
+ * All components are normalized in the floating-point range [0.0, 1.0].
+ * If none of the above formats can be detected, all components are
+ * instead set to zero.
+ *
+ * @param {String} str - the string to parse the color information from
+ * @returns {Object} an object containing the four color components,
+ *                   the structure of which is:
+ *                   { r : <red>, g : <green>, b : <blue>, a : <alpha> },
+ *                   where each component is normalized in the range
+ *                   [0.0, 1.0]; failure to match a color format yields
+ *                   zero for all components
+ */
 function _colorParse ( str )
```

Comment:
Check usage. Perhaps method of SFColor.

Reply:
This global function is only used once, farther below, in the codebase, in SFColor.colorParse.  I am not sure why it was done this way.

Reply:
The function ``_colorParse'' is actually referred to twice in the source code, once in ``x3dom.fields.SFColor.colorParse'', another time in ``x3dom.fields.SFColorRGBA.colorParse''.

Reply:
Good find. Well, something to deal with another time.